lambda/translmod.ml

```diff
@@ -1582,13 +1596,15 @@ let close_toplevel_term (lam, ()) =
 
 let transl_toplevel_item ~scopes item =
   match item.str_desc with
-    Tstr_eval (expr, _)
+    (* These first two cases are special compilation for toplevel "let _ =
+       expr", so that Toploop can display the result of the expression.
+       Otherwise, the normal compilation would result in a Lsequence returning
+       unit. *)
+    Tstr_eval (expr, layout, _) ->
+      layout_must_be_value expr.exp_loc expr.exp_type layout;
```

Comment:
Does this really need to be value? I would expect a "make sure it's not `void`" check here, in line with the check for `;` expressions

Reply:
Good point!  As a result, transl was incorrectly rejecting:

```
module M = struct
  assert false
end
```

I've fixed this (and the two other places where `Tstr_eval` is handled)